app/scripts/components/analysis/results/use-analysis-params.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useState } from 'react';
+import qs from 'qs';
+import { useLocation } from 'react-router';
+import { Feature, MultiPolygon } from 'geojson';
+import { DatasetLayer, datasets as deltaDatasets } from 'delta/thematics';
+
+import { utcString2userTzDate } from '$utils/date';
+import { polygonUrlDecode } from '$utils/polygon-url';
+
+// ?start=2015-01-01T00:00:00.000Z&end=2022-01-01T00:00:00.000Z&datasets=no2-monthly|blue-tarp-planetscope&aoi=-9.60205,36.72127|-7.03125,36.87962|-6.85546,39.45316|-6.52587,40.93011|-5.55908,42.11452|-9.38232,42.22851|-8.89892,40.84706|-9.93164,38.85682|-9.47021,38.08268|-8.96484,38.09998||-12.01904,25.95804|-8.65722,25.97779|-8.67919,27.68352|-13.05175,27.68352|-14.89746,25.95804|-12.04101,24.44714
+
+type AnalysisParams =
+  | {
+      date: {
+        start: Date;
+        end: Date;
+      };
+      datasetsLayers: DatasetLayer[];
+      aoi: Feature<MultiPolygon>;
+      errors: null
+    }
+  | {
+      date: {
+        start: null;
+        end: null;
+      };
+      datasetsLayers: null;
+      aoi: null;
+      errors: null | any[];
+    };
+
+const initialState = {
+  date: {
+    start: null,
+    end: null
+  },
+  datasetsLayers: null,
+  aoi: null,
+  errors: null
+};
+
+export function useAnalysisParams(): AnalysisParams {
+  const location = useLocation();
+
+  const [params, setParams] = useState<AnalysisParams>(initialState);
+
+  useEffect(() => {
+    const { start, end, datasets, aoi } = qs.parse(location.search, {
+      ignoreQueryPrefix: true
+    });
+
+    try {
+      if (!start || !end || !datasets || !aoi) {
+        throw [
+          'Missing required value from URL:',
+          {
+            start,
+            end,
+            datasets,
+            aoi
+          }
+        ];
+      }
+
+      const startDate = utcString2userTzDate(start);
+      const endDate = utcString2userTzDate(end);
+      if (isNaN(startDate.getTime()) || isNaN(endDate.getTime())) {
+        throw [
+          'Invalid start or end date:',
+          {
+            start,
+            end
+          }
+        ];
+      }
+
+      // Create an array with all the dataset layers.
+      const allDatasetLayers = Object.values(deltaDatasets).flatMap(
```

Comment:
Is it ok to aggregate all the bands? (I am assuming DeltaDatasets come as an object with band index as key like {1: []..2:[]..}. please correct me if I am wrong!)  I was not sure if we need to treat each band separately or not.

Reply:
Actually I was just returning the data from band 1.
I do not know if there's a specific band to use, but since in some datasets we're using band index 1, I defaulted to that.